packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/SentenceScopeHandler/SentenceSegmenter.ts

```diff
@@ -0,0 +1,60 @@
+import * as sbd from "sbd";
+import { MatchedText } from "../../../../util/regex";
+
+// A sentence starts with a letter
+const leadingOffsetRegex = /\p{L}/u;
+// A line with no letters is invalid and breaks sentences
+const invalidLineRegex = /(\n[^\p{L}]*\n)/gu;
+
+const options: sbd.Options = {
+  ["newline_boundaries"]: false,
+  ["preserve_whitespace"]: true,
+};
+
+export default class SentenceSegmenter {
+  *segment(text: string): Iterable<MatchedText> {
+    const sentences = sbd.sentences(text, options);
+    let index = 0;
+
+    for (const sentence of sentences) {
+      const parts = sentence.split(invalidLineRegex);
+
+      for (const part of parts) {
+        if (!isInvalidLine(part)) {
+          const segment = createSegment(part, index);
+          if (segment != null) {
+            yield segment;
+          }
+        }
+
+        index += part.length;
+      }
+    }
+  }
+}
+
+function createSegment(
+  sentence: string,
+  index: number,
+): MatchedText | undefined {
+  const leadingOffset =
+    sentence.match(leadingOffsetRegex)?.index ?? sentence.length;
+
+  if (sentence.length === leadingOffset) {
```

Comment:
I'm nitpicking here, but I'd prob just drop the `?? sentence.length` above and check if `leadingOffset` is `null` here.  I'd usually just make this change myself but I don't want to lose my other comment on that line 😅